src/v/kafka/server/group.cc

```diff
@@ -1244,6 +1276,28 @@ group::store_offsets(offset_commit_request&& r) {
       });
 }
 
+ss::future<txn_offset_commit_response>
+group::handle_txn_offset_commit(txn_offset_commit_request&& r) {
+    if (in_state(group_state::dead)) {
+        co_return txn_offset_commit_response(
+          r, error_code::coordinator_not_available);
+    } else if (in_state(group_state::empty)) {
+        // <kafka>The group is only using Kafka to store offsets.</kafka>
+        co_return co_await store_txn_offsets(std::move(r));
+    } else if (
+      in_state(group_state::stable)
+      || in_state(group_state::preparing_rebalance)) {
+        co_return co_await store_txn_offsets(std::move(r));
+    } else if (in_state(group_state::completing_rebalance)) {
+        co_return txn_offset_commit_response(
+          r, error_code::rebalance_in_progress);
+    } else {
```

Comment:
no need for the last `else` it's the default, so you can just return that.

Reply:
agree but it follows the pattern set by `handle_leave_group`, and `handle_offset_commit` and I don't want to introduce code which style isn't consistent with the surrounding so leaving as is